source/common/modules/markdown-editor/hooks/note-preview.js

```diff
@@ -0,0 +1,95 @@
+const tippy = require('tippy.js').default
+const { ipcRenderer } = require('electron')
+const { DateTime } = require('luxon')
+/**
+ * A hook for displaying link tooltips which display metadata
+ * and content of a file
+ *
+ * @param   {CodeMirror}  cm  The instance to attach to
+ */
+module.exports = (elem) => {
+  elem.getWrapperElement().addEventListener('mousemove', (event) => {
+    let a = event.target
+
+    // Only for note links
+    if (!a.classList.contains('cm-zkn-link')) {
+      return
+    }
+
+    // If there's already a tippy on the instance, don't re-render it
+    if (a.hasOwnProperty('_tippy')) {
+      return
+    }
+    // TODO: Translate!
+    // Initialise displayed attributes to 'null'
+    let title = 'null'
+    let content = 'null'
+    let wordCount = 'null'
+    let time = 'null'
+    // Create a tippy. This will display the loading values
+    let tooltip = tippy(a, {
+      content: 'Searching For File...',
+      allowHTML: true, // Obviously
+      interactive: true,
+      placement: 'top-start', // Display at the beginning of the anchor
+      appendTo: document.body, // anchor
+      showOnCreate: true, // Immediately show the tooltip
+      arrow: false // No arrow for these tooltips
+    })
+
+    // Find the file's absolute path
+    ipcRenderer.invoke('application', { command: 'file-find-and-return', payload: a.innerText })
+      .then((descriptorWithContent) => {
+        // If the file is found
+        if (descriptorWithContent !== null) {
+          // Get the contents of the file such as:
+          // 4 lines of 50
+          for (let i = 0; i < 4; i++) {
+            content = descriptorWithContent.content.substring(0 * i, 50 * i)
+            // prepare a newline if needed for the next loop
+            if (descriptorWithContent.content.length > 50 * i) {
+              content += '\n'
+            } else {
+              break
+            }
+          }
+          if (content.length > 200) {
+            content += '...'
+          }
+          wordCount = descriptorWithContent.wordCount // The word count
+          title = descriptorWithContent.name // The file name
+
+          // use luxon to get a local time difference
+          let dateDif = DateTime.fromMillis(descriptorWithContent.modtime).diffNow([ 'days', 'hours', 'minutes', 'seconds' ]).toObject()
+
+          // Display this using top down logic, i.e. use days, and if not, hours, then minutes, then just now
+          if (dateDif['days'] * -1 >= 1) {
+            time = Math.floor(dateDif['days'] * -1) + ' Day'
+            if (dateDif['days'] * -1 > 1) {
+              time += 's'
+            }
+            time += ' ago'
```

Comment:
Alright, you're using Luxon, but you're still doing weird stuff. Point taken, the Luxon docs are hard, but have you tried this function (or something along these lines)?

https://github.com/Zettlr/Zettlr/blob/develop/source/common/util/format-date.js

Reply:
I did try to do that, but for some reason I can't get the function into my code. Not sure what im doing wrong, as all the found usages use the 'import' method, whereas I have to use the require method. I am not sure why this like isnt working:

const { formatDate } = require('../../../util/format-date.js')

and then running it as:

formatDate(dateDif)

I have tried to research this, but I can't find the words to describe exactly what I'm doing.

Reply:
You should be able to use `import { formatDate } from '../../../util/format-date'`

Requires are deprecated either way.

Additionally, you need to pass a single Date object (or milliseconds; you can pass `descriptorWithContent.modtime` to that function), because `Date.now()` will automatically be instantiated within the function. You do _not_ need to pass a timedelta.